### ISSUE TITLE
"export * as" false positives for "import/export"

### ISSUE THREAD

**Issue by bradennapier** [benmosher/eslint-plugin-import#1803]
When you have multiple files that have an identical signature and use `export * as ___`, the `import/export` function fails to handle it properly and understand the namespaces.

### The Problem 

- say each file in question exports a "someFunction" function.  

```typescript
export someFunction() {}
```

- we have an index.js file which exports them all

```typescript
export * as one from './one';
export * as two from './two';
```

then we will get an error: `Multiple exports of name 'someFunction'. eslint(import/export)`

### The Solution

The plugin should view the above as-if it were:

```typescript
import * as one from './one';
import * as two from './two';

export { one, two };
```

### Dependency Versions

```json
"eslint": "^7.2.0",
"eslint-plugin-import": "^2.20.2",
```

**Comment by ljharb** [comment]
Can you confirm if you get this error when using the latest master branch of this plugin?

**Comment by bradennapier** [comment]
Nope, seems fixed!  I hope that was jsut released because I can swear I made sure to check, but thanks !

**Comment by ljharb** [comment]
Yes, it was released yesterday :-) thanks!
